src/GraphQL.Tests/Instrumentation/OpenTelemetryTests.cs

```diff
@@ -0,0 +1,384 @@
+        var result = await _executer.ExecuteAsync(executionOptions).ConfigureAwait(false);
+
+        // verify GraphQL response
+        _serializer.Serialize(result).ShouldBe("""{"data":{"hello":"World"}}""");
+
+        // verify activity telemetry
+        _exportedActivities.ShouldBeEmpty();
+        ranFilter.ShouldBeTrue();
+    }
+
+    [Fact]
+    public async Task WithValidationError()
+    {
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(new ExecutionOptions
+        {
+            Query = "query helloQuery { hello { dummy } }",
+            RequestServices = _host.Services,
+        }).ConfigureAwait(false);
+
+        // verify GraphQL response
+        result.Errors.ShouldNotBeNull().Count.ShouldBeGreaterThan(0);
+
+        // verify activity telemetry
+        var activity = _exportedActivities.ShouldHaveSingleItem();
+        activity.Tags.ShouldBe(new KeyValuePair<string, string>[]
+        {
+            new("graphql.document", "query helloQuery { hello { dummy } }"),
+            new("graphql.operation.type", "query"),
+            new("graphql.operation.name", "helloQuery"),
+        });
+        activity.DisplayName.ShouldBe("query helloQuery");
+#if NET6_0_OR_GREATER
+        activity.Status.ShouldBe(ActivityStatusCode.Unset);
+#endif
+    }
+
+    [Fact]
+    public async Task WithParseError()
+    {
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(new ExecutionOptions
+        {
+            Query = "{",
+            RequestServices = _host.Services,
+        }).ConfigureAwait(false);
+
+        // verify GraphQL response
+        result.Errors.ShouldNotBeNull().Count.ShouldBeGreaterThan(0);
+
+        // verify activity telemetry
+        var activity = _exportedActivities.ShouldHaveSingleItem();
+        activity.Tags.ShouldBe(new KeyValuePair<string, string>[]
+        {
+            new("graphql.document", "{"),
+        });
+        activity.DisplayName.ShouldBe("graphql");
+#if NET6_0_OR_GREATER
+        activity.Status.ShouldBe(ActivityStatusCode.Unset);
+#endif
+    }
+
+    [Fact]
+    public async Task WithServerError()
+    {
+        // execute GraphQL document
+        var result = await _executer.ExecuteAsync(new ExecutionOptions
+        {
+            Query = "{ serverError }",
+            RequestServices = _host.Services,
+        }).ConfigureAwait(false);
+
+        // verify GraphQL response
+        result.Errors.ShouldNotBeNull().Count.ShouldBeGreaterThan(0);
+
+        // verify activity telemetry
+        var activity = _exportedActivities.ShouldHaveSingleItem();
+        activity.Tags.ShouldBe(new KeyValuePair<string, string>[]
+        {
+            new("graphql.document", "{ serverError }"),
+            new("graphql.operation.type", "query"),
+#if !NET6_0_OR_GREATER
+            new("otel.status_code", "ERROR"),
+#endif
+        });
+        activity.DisplayName.ShouldBe("query");
+#if NET6_0_OR_GREATER
```

Comment:
For tests I would add polyfill to not bother with ifdefs everywhere:
```c#
internal static class ActivityHelperExtensions
{
    private const string UnsetStatusCodeTagValue = "UNSET";
    private const string OkStatusCodeTagValue = "OK";
    private const string ErrorStatusCodeTagValue = "ERROR";

    private static ActivityStatusCode? GetStatusCodeForTagValue(string statusCodeTagValue)
    {
        return statusCodeTagValue switch
        {
            string _ when UnsetStatusCodeTagValue.Equals(statusCodeTagValue, StringComparison.OrdinalIgnoreCase) => ActivityStatusCode.Unset,
            string _ when ErrorStatusCodeTagValue.Equals(statusCodeTagValue, StringComparison.OrdinalIgnoreCase) => ActivityStatusCode.Error,
            string _ when OkStatusCodeTagValue.Equals(statusCodeTagValue, StringComparison.OrdinalIgnoreCase) => ActivityStatusCode.Ok,
            _ => null,
        };
    }

    private static bool TryGetStatusCodeForTagValue(string statusCodeTagValue, out ActivityStatusCode statusCode)
    {
        ActivityStatusCode? tempStatusCode = GetStatusCodeForTagValue(statusCodeTagValue);

        statusCode = tempStatusCode ?? default;

        return tempStatusCode.HasValue;
    }

    /// <summary>
    /// Gets the status of activity execution.
    /// Activity class in .NET does not support 'Status'.
    /// This extension provides a workaround to retrieve Status from special tags with key name otel.status_code and otel.status_description.
    /// </summary>
    /// <param name="activity">Activity instance.</param>
    /// <param name="statusCode"><see cref="StatusCode"/>.</param>
    /// <param name="statusDescription">Status description.</param>
    /// <returns><see langword="true"/> if <see cref="Status"/> was found on the supplied Activity.</returns>
    public static bool TryGetStatus(this Activity activity, out ActivityStatusCode statusCode, out string statusDescription)
    {
        Debug.Assert(activity != null, "Activity should not be null");

        bool foundStatusCode = false;
        statusCode = default;
        statusDescription = null;

        foreach (ref readonly var tag in activity.EnumerateTagObjects())
        {
            switch (tag.Key)
            {
                case "otel.status_code":
                    foundStatusCode = TryGetStatusCodeForTagValue(tag.Value as string, out statusCode);
                    if (!foundStatusCode)
                    {
                        // If status code was found but turned out to be invalid give up immediately.
                        return false;
                    }

                    break;
                case "otel.status_description":
                    statusDescription = tag.Value as string;
                    break;
                default:
                    continue;
            }

            if (foundStatusCode && statusDescription != null)
            {
                // If we found a status code and a description we break enumeration because our work is done.
                break;
            }
        }

        return foundStatusCode;
    }

    public static ActivityStatusCode Status(this Activity activity)
    {
#if NET6_0_OR_GREATER
        return activity.Status;
#else
        return activity.TryGetStatus(out ActivityStatusCode statusCode, out string statusDescription)
            ? statusCode
            : ActivityStatusCode.Unset;
#endif
    }
}
```

So we can use `activity.Status().ShouldBe(ActivityStatusCode.Error);`

The code above was copied from `OpenTelemetry.Trace.ActivityHelperExtensions`.

Reply:
You could.  I'm not.  It is simpler as-is with a couple ifdefs.

Reply:
ok